src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
+
+#include <algorithm>
```

Comment:
Maybe call this file "cata_algorithm.h" just to avoid shadowing the standard library header?  I know <> vs "" should prevent problems, but it still makes me a bit twitchy.